src/compiler/mod.rs

```diff
@@ -184,11 +186,20 @@ impl Visitor<Value> for Compiler {
             Value::Bool(b) => {
                 self.builder.build_cond_br(&b, &then_block, &else_block);
                 self.builder.position_builder_at_end(&then_block);
+
+                let mut is_break = false;
                 for stmt in &expr.body {
                     self.walk(stmt);
+
+                    if matches!(stmt, expression::Expression::Break) {
```

Comment:
couldn't we just have?
```
if matches!(stmt, expression::Expression::Break) {
  self.builder.create_br(&after_if_block);
  brak;
}
```

Reply:
not exactly because after loop block branch should be created only when in iteration will NOT meet break statement.

Reply:
yeah, that is indeed the case in the second occurrence of this - you've added similar code 2 times in this PR. I think you are describing the other  occurrence, here the ``is_break``` variable seems to be only used once :)